File: backend/FinTech.Core/Entities/Loan.cs
using FinTech.Core.Common;
using FinTech.Core.Enums;

namespace FinTech.Core.Entities;

public class Loan : BaseEntity<Guid>
{
    public string UserId { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public int Term { get; set; }
    public decimal InterestRate { get; set; }
    public LoanType LoanType { get; set; }
    public LoanStatus Status { get; set; }
    public decimal MonthlyPayment { get; set; }
    public decimal MonthlyIncome { get; set; }
    public ICollection<PaymentSchedule> PaymentSchedules { get; set; } = new List<PaymentSchedule>();
    public ICollection<Transaction> Transactions { get; set; } = new List<Transaction>();
}

File: frontend/src/types/loan.ts
export type LoanStatus = 'Pending' | 'Approved' | 'Rejected' | 'Active'
export type LoanType = 'Fixed' | 'Decreasing'

export interface Loan {
  id: string
  userId: string
  amount: number
  term: number
  interestRate: number
  loanType: LoanType
  status: LoanStatus
  monthlyPayment: number
  createdAt: string
}

export interface PaymentSchedule {
  paymentNumber: number
  dueDate: string
  totalPayment: number
  principal: number
  interest: number
  remainingBalance: number
  status: 'Pending' | 'Paid'
}

export interface SimulateRequest {
  amount: number
  term: number
  loanType: LoanType
}
